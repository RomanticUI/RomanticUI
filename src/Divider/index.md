---
nav: 组件
group: 通用
mobile: false
toc: content
---

# Divider

## 代码演示

## API

| 属性 | 说明 | 类型 | 默认值 |
| ---- | ---- | ---- | ------ |
|      |      |      |        |

## Divider

Demo:

```tsx
import React from 'react';
import { Divider } from 'RomanticUI';

export default () => <Divider content={'text'} dashed={true} />;
```
