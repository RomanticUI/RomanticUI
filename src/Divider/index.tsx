import React, { Component } from 'react';
import './style/index.less';

export interface DividerProps {
  prefixCls?: string;
  content?: string;
  roClass?: string;
  textStyle?: object;
  dashed?: boolean;
}

class Divide extends Component<DividerProps> {
  state = {};

  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
  render() {
    const { prefixCls, content, roClass, dashed, ...restProps } = this.props;
    let divider;
    const lineStyle = {
      class1: dashed === true ? 'ro-divider-outer-dashed' : 'ro-divider-outer-undashed',
    };
    console.log(lineStyle.class1, 'dashed');
    // eslint-disable-next-line prefer-const
    divider = (
      <div className={`ro-divider-outer`}>
        <div className={`ro-divider-line ${lineStyle.class1}`}></div>
        {content ? <div className="ro-divider-content">{content}</div> : ''}
      </div>
    );

    return (
      <div {...restProps} className="ro-divider">
        {' '}
        {divider}
      </div>
    );
  }
}

export default Divide;
