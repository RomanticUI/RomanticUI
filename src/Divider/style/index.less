.ro-divider {
  &-outer {
    display: flex;
    align-items: center;
    width: 100%;
    margin: 16px 0;
    white-space: nowrap;
    &-undashed {
      border-block-start: 1px solid rgba(5, 5, 5, 0.06);
    }
    &-dashed {
      border-block-start: 2px dashed rgba(5, 5, 5, 0.06);
    }
  }
  &-line {
    position: relative;
    width: 95%;
    height: 2px;
    transform: translateY(50%);
    content: '';
    border-block-end: 0;
  }

  &-content {
    display: inline-block;
    width: 2em;
    padding: 0 16px;
    word-break: normal;
  }
}
