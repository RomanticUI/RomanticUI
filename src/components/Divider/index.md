---
nav: 组件
group: 通用
mobile: false
toc: content
---

# Divider

## 代码演示

## API

| 属性    | 说明      | 类型    | 默认值 |
| ------- | --------- | ------- | ------ |
| dashed  | 虚线/实现 | boolean | false  |
| content | 文字      | string  | -      |

## Divider

虚线:

```tsx
import React from 'react';
import { Divider } from 'RomanticUI';

export default () => <Divider content={'text'} dashed={true} />;
```

实线:

```tsx
import React from 'react';
import { Divider } from 'RomanticUI';

export default () => <Divider content={''} dashed={false} />;
```
