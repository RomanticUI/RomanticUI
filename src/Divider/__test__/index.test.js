// 引入测试组件
import React from 'react';
import Divider from '..';
// 使用@testing-library/react渲染对应组件
import { render } from '@testing-library/react';

describe('Divider test', () => {
  it('test case1 for dashed', () => {
    const { container } = render(<Divider content={'text'} dashed={true} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('test case1 for solid', () => {
    const { container } = render(<Divider content={''} dashed={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
