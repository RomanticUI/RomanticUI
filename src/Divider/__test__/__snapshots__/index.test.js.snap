// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider test test case1 for dashed 1`] = `
<div
  class="ro-divider"
>
   
  <div
    class="ro-divider-outer"
  >
    <div
      class="ro-divider-line ro-divider-outer-dashed"
    />
    <div
      class="ro-divider-content"
    >
      text
    </div>
  </div>
</div>
`;

exports[`Divider test test case1 for solid 1`] = `
<div
  class="ro-divider"
>
   
  <div
    class="ro-divider-outer"
  >
    <div
      class="ro-divider-line ro-divider-outer-undashed"
    />
  </div>
</div>
`;
